src/cast_core/test/KafkaLoader_test.cpp

```diff
@@ -0,0 +1,62 @@
+// Copyright 2023-present MongoDB Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <bsoncxx/json.hpp>
+#include <bsoncxx/builder/stream/document.hpp>
+
+#include <boost/exception/diagnostic_information.hpp>
+
+#include <yaml-cpp/yaml.h>
+
+#include <testlib/KafkaTestFixture.hpp>
+#include <testlib/ActorHelper.hpp>
+#include <testlib/helpers.hpp>
+
+#include <gennylib/context.hpp>
+
+namespace genny {
+namespace {
+using namespace genny::testing;
+namespace bson_stream = bsoncxx::builder::stream;
+
+TEST_CASE_METHOD(KafkaTestFixture, "KafkaLoader successfully connects to a Kafka broker.", "[streams][KafkaLoader]") {
```

Comment:
An Actor test case would only be run if one of the square-bracketed tags match one of the tags defined in a YAML file in [this directory](https://github.com/mongodb/genny/tree/master/src/resmokeconfig). The existing defined tags are `[sharded]`, `[single_node_replset]`, and `[three_node_replset]` (`[SelfTestActor]` is to test Genny code, not Actor code, so it should not be used). The square-bracketed tags are not arbitrary. So we have 2 options here:

1. Add one of the three existing defined tags above.
2. Add a new YAML file under the above directory to define either `[streams]` or `[KafkaLoader]`.

On a related note, I found that [this actor test case](https://github.com/mongodb/genny/blob/77f0c993ec2a0d389da9f140133cf28dcef90a42/src/cast_core/test/StreamStatsReporter_test.cpp#L34) is also not being run because neither `[streams]` nor `[StreamStatsReporter]` are defined tags. But I think this should be fixed in another PR instead of in this PR.